packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+      ).rejects.toThrow(ErrorInvalidEscrowAddressProvided);
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const trustedHandlers = [ethers.constants.AddressZero];
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(false);
+
+      await expect(
+        escrowClient.addTrustedHandlers(escrowAddress, trustedHandlers)
+      ).rejects.toThrow(ErrorEscrowAddressIsNotProvidedByFactory);
+    });
+
+    test('should throw an error if an trusted handlers length to equal 0', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const trustedHandlers = [];
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await expect(
+        escrowClient.addTrustedHandlers(escrowAddress, trustedHandlers)
+      ).rejects.toThrow(ErrorListOfHandlersCannotBeEmpty);
+    });
+
+    test('should throw an error if the trusted handlers contain invalid addresses', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const trustedHandlers = [FAKE_ADDRESS];
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await expect(
+        escrowClient.addTrustedHandlers(escrowAddress, trustedHandlers)
+      ).rejects.toThrow(new InvalidEthereumAddressError(FAKE_ADDRESS));
+    });
+
+    test('should successfully addTrustedHandlers escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const trustedHandlers = [ethers.constants.AddressZero];
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await escrowClient.addTrustedHandlers(escrowAddress, trustedHandlers);
+
+      expect(
+        escrowClient.escrowContract.addTrustedHandlers
+      ).toHaveBeenCalledWith(trustedHandlers);
+    });
+
+    test('should throw an error if the addTrustedHandlers fails', async () => {
```

fix writing:
'should throw an error if addTrustedHandlers fails'